### DISCUSSION TITLE
Limit `temp_directory` disk usage

### DISCUSSION THREAD

**Discussion by zenazn** [duckdb/duckdb#10031] [Ideas]
Watershed runs DuckDB on a variety of workloads in a production setting. In order to prevent the VM from running out of memory, we set a `memory_limit` and `temp_directory`. However, some queries use a *ton* of memory and fill up the entire disk, locking up the entire VM and causing the service to terminate.

We'd love it if DuckDB implemented a `temp_directory_limit` or similar to allow us to control DuckDB's disk usage.

**Comment by Mytherin** [comment]
A PR is up that implements this - https://github.com/duckdb/duckdb/pull/10978
